### PR TITLE
[tests-only][full-ci] test downloading old file versions

### PR DESCRIPTION
**Pull request by individual-it**
## Description
tests to check if downloading an old version of a file works correctly
the issue #36228 is demonstrated by checking the header

## Related Issue
tests for #36228

## How Has This Been Tested?
- CI here
- CI of ocis: https://github.com/owncloud/ocis/pull/2262

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/FilesVersionsContext.php [hunk 1]
```diff
@@ -52,7 +110,7 @@ class FilesVersionsContext implements Context {
 	public function userRestoresVersionIndexOfFile($user, $versionIndex, $path) {
 		$user = $this->featureContext->getActualUsername($user);
 		$fileId = $this->featureContext->getFileIdForPath($user, $path);
-		$responseXml = $this->listVersionFolder($user, "/meta/$fileId/v", 1);
+		$responseXml = $this->listVersionFolder($user, (int)$fileId, 1);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/FilesVersionsContext.php:113]:
```suggestion
		$responseXml = $this->listVersionFolder($user, $fileId, 1);
```

IMO file-id in OCIS is not always a numeric-string. Casting to `int` probably messes it up.

File: tests/acceptance/features/apiVersions/fileVersions.feature [hunk 2]
```diff
@@ -415,3 +415,41 @@ Feature: dav-versions
...
+    And the downloaded content should be "version 1"
+    When user "Alice" downloads the version of file "textfile0.txt" with the index "2"
+    Then the HTTP status code should be "200"
+    And the following headers should be set
+      | header              | value                                                                |
+      | Content-Disposition | attachment; filename*=UTF-8''textfile0.txt; filename="textfile0.txt" |
+    And the downloaded content should be "uploaded content"
+
+
+  Scenario: download an old version of a of a restored file
```
Comment (phil-davis) [inline tests/acceptance/features/apiVersions/fileVersions.feature:438]:
```suggestion
  Scenario: download an old version of a restored file
```

File: tests/acceptance/features/apiVersions/fileVersionsSharingToSharesIssue36228.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
+@api @files_versions-app-required @issue-36228 @notToImplementOnOCIS
+
+Feature: dav-versions
+
+  Background:
+    Given the administrator has set the default folder for received shares to "Shares"
+    And auto-accept shares has been disabled
+    And user "Alice" has been created with default attributes and without skeleton files
+
+  Scenario: download an old version of a of a restored file
```
Comment (phil-davis) [inline tests/acceptance/features/apiVersions/fileVersionsSharingToSharesIssue36228.feature:10]:
```suggestion
  Scenario: download old versions of a shared file as share receiver
```

**Review by phil-davis — APPROVED** [review]

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIAddUsers-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/30958/128/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIAddUsers-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/30961/128/1

**Comment by individual-it** [comment]
I don't understand where that problem comes from 

grafik

**Comment by phil-davis** [comment]
> I don't understand where that problem comes from

I have seen it happen sometimes. I think that "something happens" with the startup of mailhog, and somewhere in the handling of that (maybe in "client"  code that should talk to the email server?) "something happens" that doesn't have "happy classes" in view.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIUpload-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/30965/154/1

**Review by phil-davis — COMMENTED** [review]
Just 2 scenario descriptions to adjust.

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@individual-it squash? or keep 6 commits?

**Comment by individual-it** [comment]
@phil-davis I stopped CI and will merge your commit into one of the other commits, I feel the rest makes logical sense as it is now

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
https://drone.owncloud.com/owncloud/core/30973/150/15
Another `Fatal error: Call to undefined method GuzzleHttp\Exception\ConnectException::getResponse() (Behat\Testwork\Call\Exception\FatalThrowableError)`
I will cancel and restart the build.

**Pull request merged**